src/Controllers/ApiController.php

```diff
@@ -27,4 +39,114 @@ public function index()
             ->withHeader('content-type', 'application/json')
             ->withContent(json_encode(['error' => 'Not implemented']));
     }
+
+
+    /**
+     * @return User
+     *
+     * Warning: I am no php coder. This code isnt intended for eternity but making our lives @36c3 a bit easier.
+     * Code is mostly harvested from /includes/controller/shifts_controller.php/shifts_json_export_controller()
+     */
+    protected function doApiAuth($key)
+    {
+        $user = auth()->apiUser('key');
+        if (
+            !preg_match('/^[\da-f]{32}$/', $key)
```

Comment:
This check is useless, the user gets already authenticated over the API by using the `apiUser` method (which has the benefit that a logged in user isn't required to provide an API key)

Reply:
Just copied from shifts_json_export_controller(). 
Should be likewise removed for user convenience (after resolving this PR)

Reply:
Jeah, thats some old dead code that never got removed as changing everything thats inside the `includes/` path it is a pain